### ISSUE TITLE
mingw-w64 bottle depends on zstd but formula doesn't install it

### ISSUE THREAD

**Issue by mrpippy** [Homebrew/homebrew-core#69843]
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

Build Windows executables using mingw-w64

## What happened (include command output)

When using mingw-w64 to build anything, it fails to run because the bottle has an implicit dependency on zstd. After manually installing zstd, it does work. This is probably because of GCC 10's [zstd support for LTO byte code compression](https://gcc.gnu.org/legacy-ml/gcc/2019-07/msg00003.html), and the formula probably needs to either require zstd or pass `--with-zstd=no` when configuring gcc.

<details>
  <summary>Command output</summary>
  <pre>
 $ x86_64-w64-mingw32-gcc -dM -E - < /dev/null
dyld: Library not loaded: /usr/local/opt/zstd/lib/libzstd.1.dylib
  Referenced from: /usr/local/Cellar/mingw-w64/8.0.0_1.reinstall/toolchain-x86_64/bin/../libexec/gcc/x86_64-w64-mingw32/10.2.0/cc1
  Reason: image not found
x86_64-w64-mingw32-gcc: internal compiler error: Abort trap: 6 signal terminated program cc1
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://github.com/Homebrew/homebrew-core/issues> for instructions.
  </pre>
</details>

## What you expected to happen

mingw-w64 should work.

## Step-by-step reproduction instructions (by running `brew install` commands)

```
$ brew uninstall zstd
$ brew install mingw-w64
$ x86_64-w64-mingw32-gcc -dM -E - < /dev/null
```

**Comment by carlocab** [comment]
>if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

Please don't tick the boxes without actually doing them.

**Comment by SMillerDev** [comment]
Could you fill out the issue template please? It's pretty essential in debugging issues. This issue can be reopened once we have all the information needed to start debugging.

**Comment by mrpippy** [comment]
Ok I posted `brew config`/`doctor` logs.

**Comment by carlocab** [comment]
Thanks. This will be fixed by #69848.

**Comment by mrpippy** [comment]
#69849 triggering a bottle rebuild did solve the issue, but didn't prevent it from happening in the future. I created #70849 to fix the formula.
